conda_smithy/templates/circle.yml.tmpl

```diff
@@ -1,39 +1,31 @@
-{%- block env_branches -%}
-{%- if not matrix -%}
-general:
-  branches:
-    ignore:
-      - /.*/
+version: 2
 
-{% endif -%}
-{% endblock -%}
-{% block env_dependencies -%}
-{%- if matrix -%}
-checkout:
-  post:
-    - ./ci_support/fast_finish_ci_pr_build.sh
-    - ./ci_support/checkout_merge_commit.sh
-
-machine:
-  services:
-    - docker
-
-dependencies:
-  # Note, we used to use the naive caching of docker images, but found that it was quicker
-  # just to pull each time. #rollondockercaching
-  override:
-    - docker pull condaforge/linux-anvil
-
-{% endif -%}
-{% endblock -%}
-test:
-  override:
-{%- block env_test %}
-{%- if matrix %}
-    # Run, test and (if we have a BINSTAR_TOKEN) upload the distributions.
-    - ./ci_support/run_docker_build.sh
+jobs:
+  build:
+    working_directory: ~/test
+    machine: true
+{%- block env_test -%}
+{%- if not matrix %}
+    branches:
+      ignore:
+        - /.*/
+    steps:
+      - run:
+          # The Circle-CI build should not be active, but if this is not true for some reason, do a fast finish.
+          command: exit 0
 {% else %}
-    # The Circle-CI build should not be active, but if this is not true for some reason, do a fast finish.
-    - exit 0
-{% endif -%}
+    steps:
+      - checkout
+      - run:
+          name: Fast finish outdated PRs and merge PRs
+          command: |
+            ./ci_support/fast_finish_ci_pr_build.sh
+            ./ci_support/checkout_merge_commit.sh
+      - run:
+          command: docker pull condaforge/linux-anvil
+      - run:
+          # Run, test and (if we have a BINSTAR_TOKEN) upload the distributions.
+          command: ./ci_support/run_docker_build.sh
```

Comment:
It would be nice to check how the cache works in this new version. So we could place this command in a section which could be cached and need fewer changes to use ccache.